### PR TITLE
Add last_day_of_month function

### LINKED ISSUE #276: Add function to get the last day of the month
Ref: https://github.com/prestodb/presto/issues/8477
The old issue is labeled `syntax-needs-review`.

### PR DESCRIPTION
**Pull request by ebyhr**
Fix #276

### CODE REVIEW

File: presto-main/src/main/java/io/prestosql/operator/scalar/DateTimeFunctions.java [hunk 1]
```diff
@@ -964,6 +964,37 @@ public static long dayFromInterval(@SqlType(StandardTypes.INTERVAL_DAY_TO_SECOND
         return milliseconds / MILLISECONDS_IN_DAY;
     }
 
+    @Description("last day of the month that given timestamp")
```
Comment (electrum) [inline presto-main/src/main/java/io/prestosql/operator/scalar/DateTimeFunctions.java]: Typo, should be

> last day of the month of the given timestamp

File: presto-main/src/main/java/io/prestosql/operator/scalar/DateTimeFunctions.java [hunk 2]
```diff
@@ -964,6 +964,37 @@ public static long dayFromInterval(@SqlType(StandardTypes.INTERVAL_DAY_TO_SECOND
...
+    @ScalarFunction("last_day")
+    @SqlType(StandardTypes.DATE)
+    public static long lastDayOfMonthFromTimestampWithTimeZone(@SqlType(StandardTypes.TIMESTAMP_WITH_TIME_ZONE) long timestampWithTimeZone)
+    {
+        long millis = unpackChronology(timestampWithTimeZone).monthOfYear().roundCeiling(unpackMillisUtc(timestampWithTimeZone - MILLISECONDS_IN_DAY));
+        return MILLISECONDS.toDays(millis);
+    }
+
+    @Description("last day of the month that given timestamp")
```
Comment (electrum) [inline presto-main/src/main/java/io/prestosql/operator/scalar/DateTimeFunctions.java]: Same as above

File: presto-main/src/main/java/io/prestosql/operator/scalar/DateTimeFunctions.java [hunk 3]
```diff
@@ -964,6 +964,37 @@ public static long dayFromInterval(@SqlType(StandardTypes.INTERVAL_DAY_TO_SECOND
         return milliseconds / MILLISECONDS_IN_DAY;
     }
 
+    @Description("last day of the month that given timestamp")
+    @ScalarFunction("last_day")
+    @SqlType(StandardTypes.DATE)
+    public static long lastDayOfMonthFromTimestampWithTimeZone(@SqlType(StandardTypes.TIMESTAMP_WITH_TIME_ZONE) long timestampWithTimeZone)
+    {
+        long millis = unpackChronology(timestampWithTimeZone).monthOfYear().roundCeiling(unpackMillisUtc(timestampWithTimeZone - MILLISECONDS_IN_DAY));
```
Comment (electrum) [inline presto-main/src/main/java/io/prestosql/operator/scalar/DateTimeFunctions.java]: Maybe split this up
```java
long millis = unpackMillisUtc(timestampWithTimeZone - MILLISECONDS_IN_DAY);
millis = unpackChronology(timestampWithTimeZone).monthOfYear().roundCeiling(millis);
return MILLISECONDS.toDays(millis);

File: presto-docs/src/main/sphinx/functions/datetime.rst [hunk 4]
```diff
@@ -57,6 +57,10 @@ Date and Time Functions
 
     This is an alias for ``CAST(x AS date)``.
 
+.. function:: last_day(x) -> date
```
Comment (electrum) [inline presto-docs/src/main/sphinx/functions/datetime.rst]: Discussed with @martint and we should call this `last_day_of_month` so it's not ambiguous (with last day of year).

**Comment by ebyhr** [comment]
@electrum Thanks for your review. Updated.

**Pull request merged**